### ISSUE TITLE
Bring back market and miner tests

### ISSUE THREAD

**Issue by ZenGround0** [filecoin-project/venus#3642]
### Description
#3639 skips daemon tests related to the storage market (how client-miner deals are made, finalized and cashed out) and storage miner (how a miner interacts with the state machine to maintain storage power in the system).  These two areas are undergoing such significant changes with the new spec that it doesn't make sense to keep testing them during development.

This issue tracks the future work of removing the skips.  After the market and miner functionalities are brought back in line with the spec tests of these functionalities should either be 
(1) reinstated with any logical changes needed to test the new expected behaviors 
(2) rewritten entirely in the case the changes are so big that this is easier than rewriting the existing 
(3) removed in the case the tests hit behaviors that no longer exist or have an analogue in the new system

Rewriting tests is a good opportunity for removing dependence on the deprecated test daemon in favor of FAST.

**Comment by ZenGround0** [comment]
After #3726 we also have relevant skipped tests in `internal/pkg/protocol/storage/client_test.go`
